app/helpers/application_helper.rb

```diff
@@ -1,2 +1,116 @@
+																				 title, content, time, test)
+			# Store notification record in db for each user
+			userids = shift.users.collect(&:id)
+			userids.each do |id|
+				@user = User.find(id)
+				@user.notifications.create(notification_id: onesignal_id, start_time: time, title: title, content: content)
+			end
+			#puts @user.notifications.pluck(:start_time, :title, :content)
+			onesignal_id
+		end
+
+		# @note Helper method. Can delete if necessary but allows for additional
+		# logic for post notifications.
+		# Currently, post notifications send immediately to all users, meant to
+		# be used for line monitor announcements.
+		def post_notification(title, content, test: false)
+			# Send to all members, immediately
+			puts "Scheduling post announcement notification to all users with title: #{title} content: #{content}"
+			onesignal_id = create_notification(['All'], recipient_type='all',
+																				 title, content, time=nil, test)
+			#puts @user.notifications.pluck(:start_time, :title, :content)
+			onesignal_id
+		end
+
+    # @note Function to create a scheduled notification for a list of users
+    # by netid, which is mapped to a OneSignal device by an external ID.
+		# @param [Array] recipients
+    # @param [String] recipient_type, one of (netids, all, players)
+		# @param [String] title of notification
+		# @param [String] content of notification
+    # @param [Boolean] test parameters without hitting OneSignal API.
+		# @return [String] OneSignal ID of scheduled notification
+		def create_notification(recipients, recipient_type, title, content, time,
+														test)
+			if recipient_type == 'netids'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'include_external_user_ids' => recipients}
+			elsif recipient_type == 'all'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									 'included_segments' => ['All']}
+			elsif recipient_type == 'players'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'include_player_ids' => recipients}
+			else # assume sending to test users
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'included_segments' => ['Test Users']}
+			end
+			params['send_after'] = time if time
+			puts "Sending POST request to OneSignal with parameters: #{params}"
+			return params.to_json if test
+			uri = URI.parse('https://onesignal.com/api/v1/notifications')
+			http = Net::HTTP.new(uri.host, uri.port)
+			http.use_ssl = true
+			request = Net::HTTP::Post.new(uri.path,	'Content-Type'  => 'application/json;charset=utf-8', 'Authorization' => 'Basic NDY3ZjU0NTktZTUwNy00ODQyLWFmNTMtN2IzYjAyZjI5MGYx')
```

Comment:
Oof, no keys please in commit. This is now in our commit history, and if someone with malicious intent really wanted to, they can find it and use our key. Fixing the commit history can only be done by either remove a commit from history manually (this prob wont work with how many commits you have), squash and merging this PR, **OR** like I suggest in other places, moving to another PR and branch and removing this branch.

Reply:
The solution to having API keys is environment variables. Add the keys to the `.env` file, and let everyone on the devs team know the new key, so they can update their .env file.

Reply:
So sorry for this rookie mistake. Fixed (except for the js app id as I need to ask you how to load env variables into the js). Sent you the updated .env file.